test/e2e/e2eutil/pravegacluster_util.go

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package e2eutil
+
+import (
+	goctx "context"
+	"fmt"
+	"testing"
+	"time"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	api "github.com/pravega/pravega-operator/pkg/apis/pravega/v1alpha1"
+	util "github.com/pravega/pravega-operator/pkg/util"
+)
+
+var (
+	RetryInterval        = time.Second * 5
+	Timeout              = time.Second * 60
+	CleanupRetryInterval = time.Second * 1
+	CleanupTimeout       = time.Second * 5
+)
+
+// CreateCluster creates a PravegaCluster CR with the desired spec
+func CreateCluster(t *testing.T, f *framework.Framework, ctx *framework.TestCtx, p *api.PravegaCluster) (*api.PravegaCluster, error) {
+	t.Logf("creating pravega cluster: %v", p.Name)
+	err := f.Client.Create(goctx.TODO(), p, &framework.CleanupOptions{TestContext: ctx, Timeout: CleanupTimeout, RetryInterval: CleanupRetryInterval})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create CR: %v", err)
+	}
+
+	// Give the operator a few seconds to fill the PravegaCluster spec with default values if necessary
+	time.Sleep(5 * time.Second)
```

Comment:
I am wondering what if the cluster is very slow? I see some examples are first using waitForPravegaCluster to make sure the cluster is running, then getting CR will be reliable.

Reply:
This line is not waiting for the Pravega cluster to come up, it's just giving a few seconds for the operator to process the CR and fill the missing parameters with default values. 5 seconds should be more than enough for the operator to do it. However, I'm open to other, more reliable alternatives.

Reply:
We should be using a mutating webhook to apply defaults, so that they are applied synchronously.

Reply:
@EronWright that will happen when we implement an [Admission Controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/) to validate CR objects and apply default values. This work is tracked here: https://github.com/pravega/pravega-operator/issues/7